modules/rosa-hcp-vpc-terraform.adoc

```diff
@@ -0,0 +1,68 @@
+// Module included in the following assemblies:
+//
+// * rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-quickly.adoc
+// * rosa_getting_started/quickstart.adoc
+
+:_content-type: PROCEDURE
+[id="rosa-hcp-vpc-terraform_{context}"]
+= Creating an AWS Virtual Private Cloud using Terraform
+
+Terraform is a tool that allows you to create various resources using an established template. The following process uses the default options as required to create a {hcp} for ROSA cluster. For more information about using Terraform, see the additional resources.
+
+.Prerequisites
+
+* You have installed Terraform on your machine.
+
+.Procedure
+
+. Open a shell prompt and create a directory for your terraform files by running the following command:
++
+[source,terminal]
+----
+$ mkdir hypershift-tf
+----
+
+. Navigate to the created directory by running the following command:
++
+[source,terminal]
+----
+$ cd hypershift-tf
+----
+
+
+. Download the VPC setup terraform file by running the following command:
++
+[source,terminal]
+----
+$ curl -s -o setup-vpc.tf https://raw.githubusercontent.com/openshift-cs/OpenShift-Troubleshooting-Templates/master/rosa-hcp-terraform/setup-vpc.tf
+----
+
+. After the template has downloaded, initiate the terraform file by running the following command. You see a message confirming the initialization when this process completes: 
++
+[source,terminal]
+----
+$ terraform init
+----
+
+. To build your VPC terrafom plan based off of the downloaded template, run the `plan` command. You may specify a cluster name and your AWS region.
++
+[source,terminal]
+----
+$ terraform plan -out rosa.plan [-var aws_region=<region>] [-var cluster_name=<cluster_name>]
+----
+
+. You should have a `rosa.plan` file in the directory that you created in the first step. Apply this plan file to build your VPC by running the following command:
++
+[source,terminal]
+----
+$ terraform apply rosa.plan
+----
+
+. Optional: You can capture the values of the terraform-provisioned private, public, and machinepool subnet IDs as environment variables to use when creating your {hcp} for ROSA cluster by running the following command:
++
+[source,terminal]
+----
+$ PRIVATE_SUBNET=`terraform output -raw cluster-private-subnet`
+$ PUBLIC_SUBNET=`terraform output -raw cluster-public-subnet`
+$ MACHINEPOOL_SUBNET=`terraform output -raw node-private-subnet`
+----
```

Comment:
Technically, there should be three substeps here (only one command is allowed in one step). Something like:

```suggestion
. Optional: You can capture the values of the Terraform-provisioned private, public, and machinepool subnet IDs as environment variables to use when creating your {hcp} for ROSA cluster by running the following commands:

** For the private subnet ID:
+
[source,terminal]
----
$ PRIVATE_SUBNET=`terraform output -raw cluster-private-subnet`
----

** For the public subnet ID:
+
[source,terminal]
----
$ PUBLIC_SUBNET=`terraform output -raw cluster-public-subnet`
----

** For the machinepool subnet ID:
+
[source,terminal]
----
$ MACHINEPOOL_SUBNET=`terraform output -raw node-private-subnet`
----
```